src/components/PeoplePage.tsx

```diff
@@ -1,31 +1,143 @@
+  }
+
+  useEffect(() => {
+    loadPeople();
+  }, []);
+
+  const [searchParams] = useSearchParams();
+  const sort = searchParams.get('sort') || '';
+  const order = searchParams.get('order') || '';
+  const query = searchParams.get('query') || '';
+  const sex = searchParams.get('sex') || '';
+  const centuries = searchParams.getAll('centuries') || [];
+
+  function getPreparedPeople() {
+    let result = [...people];
+
+    if (query) {
+      result = people.filter(
+        (person: Person) => person.name.toLowerCase()
+          .includes(query.toLowerCase())
+          || (person.motherName
+            ? person.motherName.toLowerCase().includes(query.toLowerCase())
+            : false)
+          || (person.fatherName
+            ? person.fatherName.toLowerCase().includes(query.toLowerCase())
+            : false),
+      );
+    }
+
+    if (sex) {
+      result = result.filter((person) => {
+        switch (sex) {
+          case 'm':
+            return person.sex === 'm';
+
+          case 'f':
+            return person.sex === 'f';
+
+          default:
+            return true;
+        }
+      });
+    }
+
+    if (centuries.length) {
+      let peopleOfCentyries: Person[] = [];
+
+      for (let i = 16; i < 21; i += 1) {
+        if (centuries.includes(`${i}`)) {
+          peopleOfCentyries = [
+            ...peopleOfCentyries,
+            ...result.filter(
+              (person: Person) => Math.ceil(person.born / 100) === i,
+            ),
+          ];
+        }
+      }
+
+      result = peopleOfCentyries;
+    }
+
+    if (sort) {
+      result.sort((p1, p2) => {
+        switch (sort) {
+          case 'name':
+          case 'sex':
+            return p1[sort].localeCompare(p2[sort]);
+          case 'born':
+          case 'died':
+            return p1[sort] - p2[sort];
+          default:
+            return 1;
+        }
+      });
+    }
+
+    if (order) {
+      return result.reverse();
+    }
+
+    return result;
+  }
+
+  const preparedPeople = getPreparedPeople();
+
   return (
     <>
       <h1 className="title">People Page</h1>
 
       <div className="block">
         <div className="columns is-desktop is-flex-direction-row-reverse">
           <div className="column is-7-tablet is-narrow-desktop">
-            <PeopleFilters />
+            {!isLoading && <PeopleFilters />}
           </div>
 
           <div className="column">
             <div className="box table-container">
-              <Loader />
+              {isLoading && <Loader />}
 
-              <p data-cy="peopleLoadingError">Something went wrong</p>
+              {error && (
+                <p data-cy="peopleLoadingError" className="has-text-danger">
+                  Something went wrong
+                </p>
+              )}
 
-              <p data-cy="noPeopleMessage">
-                There are no people on the server
-              </p>
+              {people.length === 0 && !isLoading && (
```

Comment:
```suggestion
              {!people.length && !isLoading && (
```
**PLS DO THE SAME FOR ALL SUCH CASES IN THE PROJECT**